cinema/views.py

```diff
@@ -97,12 +100,30 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
+        elif self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
```

Comment:
The `get_serializer_class` method should handle the case where `self.action` is not one of the specified actions ('list', 'retrieve', 'upload_image'). Ensure that the default serializer class is appropriate for other actions.